cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -149,16 +149,29 @@ func CreateStackedEtcdStaticPodManifestFile(client clientset.Interface, manifest
 		return err
 	}
 
-	// notifies the other members of the etcd cluster about the joining member
 	etcdPeerAddress := etcdutil.GetPeerURL(endpoint)
 
-	klog.V(1).Infof("Adding etcd member: %s", etcdPeerAddress)
-	initialCluster, err := etcdClient.AddMember(nodeName, etcdPeerAddress)
+	klog.V(1).Infoln("[etcd] Getting the list of existing members")
+	initialCluster, err := etcdClient.ListMembers()
 	if err != nil {
 		return err
 	}
-	fmt.Println("[etcd] Announced new etcd member joining to the existing etcd cluster")
-	klog.V(1).Infof("Updated etcd member list: %v", initialCluster)
+
+	// only add the new member if it doesn't already exists
+	klog.V(1).Infof("[etcd] Checking if the etcd member already exists: %s", etcdPeerAddress)
+	id, _ := etcdClient.GetMemberID(etcdPeerAddress)
```

Comment:
Both `ListMembers()` and `GetMemberID()` call `listMembers()`. There's no need to do that twice (especially if we are trying to solve a problem that presents itself due to loss of connectivity).
Instead, you can avoid calling `GetMemberID` by just checking if `etcdPeerAddress` matches an entry in `initialCluster`.

Reply:
that is true. i will update this later today.

Reply:
updated.